ee/desktop/runtime/runner.go

```diff
@@ -109,3 +110,68 @@ func (r *DesktopUsersProcessesRunner) Interrupt(interruptError error) {
 		}
 	}
 }
+
+// waitForProcess adds 1 to DesktopUserProcessRunner.procsWg and runs a goroutine to wait on the process to exit.
+// The go routine will decrement DesktopUserProcessRunner.procsWg when it exits
+func (r *DesktopUsersProcessesRunner) waitOnProcessAsync(uid string, proc *os.Process) {
```

Comment:
Is the intent here to track the processes per user?

Though somewhat elegant, one downside here is that we cannot track which process is stuck. Might still be an okay first pass.

Reply:
The intent here is to just wait on the process so that if it does die, the parent can clean it up and we don't get a zombie. If you wanted to test this you could not call this function then terminate the desktop process, a new one will not spin up because this one remains alive or at least undead.

I've updated the function comment with more info

The wait group was to satisfy the race detector.